File: algorithms/1/l1/크레인 인형뽑기 게임.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
    int answer = 0;
    vector<int> stack;
    int n = board.size();

    for(auto move: moves){
        int idx = move-1;
        for(int i=0; i<n; i++){
            if(board[i][idx]!=0){
                stack.push_back(board[i][idx]);
                board[i][idx] = 0;
                break;
            }
        }
        
        if(stack.empty() || stack.size()==1) continue;
        else if (stack[stack.size()-1] == stack[stack.size()-2]){
            stack.erase(stack.end()-1);
            stack.erase(stack.end()-1);
            answer+=2;
        } 
    }
    return answer;
}

File: algorithms/1/l1/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    n = len(board)
    stack =[]
    for move in moves:
        idx = move - 1
        for i in range(n):
            if board[i][idx] != 0:
                stack.append(board[i][idx])
                board[i][idx] = 0
                break
        
        if len(stack) == 0 or len(stack) ==1:
            continue
        elif stack[-2] == stack[-1]:
            stack.pop(-1)
            stack.pop(-1)
            answer+=2
        else:
            continue

    return answer